lib/src/rules/always_specify_types.dart

```diff
@@ -114,8 +123,16 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitDeclaredIdentifier(DeclaredIdentifier node) {
-    if (node.type == null) {
-      rule.reportLintForToken(node.keyword);
+    var keyword = node.keyword;
+    if (node.type == null && keyword != null) {
+      var element = node.identifier.staticElement;
+      if (element is VariableElement) {
+        var description = keyword.keyword == Keyword.VAR
+            ? "'$keyword' could be '${element.type}'."
+            : "Specify '${element.type}' type.";
```

Comment:
By convention, the correction messages all follow the pattern "Try <solution>", so I'd propose a message like "Try specifying the type '{0}'.". That doesn't specify whether to replace the keyword so we wouldn't need the extra test.

Similarly below.